### ISSUE TITLE
gtest and libcmaes + NLopt and other doc issues

### ISSUE THREAD

**Issue by JulesKozolinsky** [resibots/limbo#117]
- use ldconfig after installation of libcmaes and gtest
- do not use debian package for NLopt

**Comment by costashatz** [comment]
Thank you for using our code.

> use ldconfig after installation of libcmaes and gtest

It is common that sometimes `ldconfig` is needed after manual installation of libraries. We could add a hint in the [docs](http://www.resibots.eu/limbo/tutorials/compilation.html). Thanks.

> do not use debian package for NLopt
 
I believe that int the [docs](http://www.resibots.eu/limbo/tutorials/compilation.html) we are clearly stating to not use the debian package for NLOpt, but compile it from source.


Please refer to the documentation for more details on installing and using limbo: http://www.resibots.eu/limbo/index.html

**Comment by jbmouret** [comment]
I told him to add this as an issue to remind us to make the doc clearer/
update.

Jules: your message could have been clearer...

**Comment by JulesKozolinsky** [comment]
I had some issues concerning the **basic example in the documentation's tutorial.** 

The following code ` ./waf build --exp test ` gave me the following error : 

```
In file included from /home/jules/Documents/limbo/src/limbo/bayes_opt/bo_base.hpp:26:0,
                 from /home/jules/Documents/limbo/src/limbo/bayes_opt/boptimizer.hpp:14,
                 from ../exp/test/main.cpp:2:
/home/jules/Documents/limbo/src/limbo/kernel/squared_exp_ard.hpp: In instantiation of ‘limbo::kernel::SquaredExpARD<Params>::SquaredExpARD(int) [with Params = Params]’:
/home/jules/Documents/limbo/src/limbo/model/gp.hpp:26:44:   required from ‘limbo::model::GP<Params, KernelFunction, MeanFunction, HyperParamsOptimizer>::GP() [with Params = Params; KernelFunction = limbo::kernel::SquaredExpARD<Params>; MeanFunction = limbo::mean::Data<Params>; HyperParamsOptimizer = limbo::model::gp::KernelLFOpt<Params, limbo::opt::ParallelRepeater<Params, limbo::opt::Rprop<Params> > >]’
/home/jules/Documents/limbo/src/limbo/bayes_opt/boptimizer.hpp:65:15:   required from here
/home/jules/Documents/limbo/src/limbo/kernel/squared_exp_ard.hpp:31:103: error: no type named ‘kernel_squared_exp_ard’ in ‘struct Params’
             SquaredExpARD(int dim = 1) : _sf2(0), _ell(dim), _A(dim, Params::kernel_squared_exp_ard::k()), _input_dim(dim)
                                                                                                       ^
/home/jules/Documents/limbo/src/limbo/kernel/squared_exp_ard.hpp:34:120: error: no type named ‘kernel_squared_exp_ard’ in ‘struct Params’
                 Eigen::VectorXd p = Eigen::VectorXd::Zero(_ell.size() + _ell.size() * Params::kernel_squared_exp_ard::k() + 1);
```

**Comment by bpapaspyros** [comment]
Also it seems that the there is no package gtest as listed in the documentation (therefore sudo apt-get install gtest is not valid). However, the official ubuntu 14.04 repositories have libgtest-dev listed instead. It may not be the case for later versions of ubuntu though.

**Comment by JulesKozolinsky** [comment]
The official Ubuntu 16.04 repositories have libgtest-dev listed instead too.

**Comment by bpapaspyros** [comment]
@costashatz sudo apt-get install gtest is still present at the [documentation](http://www.resibots.eu/limbo/tutorials/compilation.html#optional-but-highly-recommended), although it is invalid. I will get back to you on the rest of the changes.

**Comment by costashatz** [comment]
> @costashatz sudo apt-get install gtest is still present at the documentation, although it is invalid. I will get back to you on the rest of the changes.

Fixed in 18e5128

**Comment by costashatz** [comment]
**Closing..**
